safekeeper/src/wal_backup.rs

```diff
@@ -74,11 +74,12 @@ pub(crate) async fn update_task(mgr: &mut Manager, need_backup: bool, state: &St
 
             let (shutdown_tx, shutdown_rx) = mpsc::channel(1);
 
-            let async_task = backup_task_main(
-                mgr.wal_resident_timeline(),
-                mgr.conf.backup_parallel_jobs,
-                shutdown_rx,
-            );
+            let Ok(resident) = mgr.wal_resident_timeline() else {
+                info!("Timeline shut down");
```

Comment:
Not this PR's fault, but when timeline manager exits it calls         `wal_backup::update_task(&mut mgr, false, &last_state).await;` while it should just directly kill the task. Probably handier to fix in this PR.

Reply:
Made this simplification in 6762fd273184da930e1cf54a56e72cab5cdba824